includes/class-wcsatt-product.php

```diff
@@ -0,0 +1,856 @@
+<?php
+/**
+ * WCS_ATT_Product API
+ *
+ * @author   SomewhereWarm <info@somewherewarm.gr>
+ * @package  WooCommerce Subscribe All the Things
+ * @since    2.0.0
+ */
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * API for working with subscription-enabled product objects.
+ *
+ * @class    WCS_ATT_Product
+ * @version  2.0.0
+ */
+class WCS_ATT_Product {
+
+	public static function init() {
+
+		// Allow WCS to recognize any product as a subscription.
+		add_filter( 'woocommerce_is_subscription', array( __CLASS__, 'filter_is_subscription' ), 10, 3 );
+
+		// Delete object meta in use by the application layer.
+		add_action( 'woocommerce_before_product_object_save', array( __CLASS__, 'delete_reserved_meta' ) );
+
+		// Make product prices scheme-dependent.
+		WCS_ATT_Scheme_Prices::add_price_filters();
+		WCS_ATT_Scheme_Prices::add_price_html_filters();
```

Comment:
@thenbrent I felt that adding these here gives developers a clearer impression of how _products_ are modified to support the subscription schemes concept. The Product API is meant for setting/getting subscription states directly -- at the same time, however, the subscription states available/set on product objects change the return values of some `WC_Product` methods via filters. Registering these filters here makes this clearer. I'm not sure if the best place for these callbacks is `WCS_ATT_Scheme_Prices` or a new `WCS_ATT_Product_Prices` class, but I wanted to keep them out of the API class.